py-polars/polars/dataframe/group_by.py

```diff
@@ -111,7 +111,13 @@ def __iter__(self) -> Self:
         return self
 
     def __next__(self) -> tuple[tuple[object, ...], DataFrame]:
-        if self._current_index >= len(self._group_indices):
+        try:
```

Comment:
Instead of this try-except, we should return an object of a new class on `__iter__`. Then it is solved by the type system.